pkg/api/types.go

```diff
@@ -642,7 +643,7 @@ type ServiceSpec struct {
 	// PortalIP is usually assigned by the master.  If specified by the user
 	// we will try to respect it or else fail the request.  This field can
 	// not be changed by updates.
-	PortalIP string `json:"portalIP,omitempty"`
+	PortalIP net.IP `json:"portalIP,omitempty"`
```

Comment:
um, let's leave the API types as string for now - there are some corner cases with JSON and net.IP that we have yet to handle (#1300)